ObserverSet.swift

```diff
@@ -26,122 +26,198 @@
 import Dispatch
 
 /// A reference to an entry in the list of observers. Use this to remove an observer.
-open class ObserverSetEntry<Parameters> {
- 
-  public typealias ObserverCallback = (Any) -> (Parameters) -> Void
-  
-  fileprivate weak var observer: AnyObject?
-  fileprivate let callBack: ObserverCallback
-  
-  fileprivate init(observer: AnyObject?, callBack: @escaping ObserverCallback) {
-    self.observer = observer
-    self.callBack = callBack
-  }
-  
-}
+public class ObserverSetEntry<Parameters> {
 
-/// A set of observers that can be notified of certain actions. A more Swift-like version of NSNotificationCenter.
-open class ObserverSet<Parameters> {
-  
-  // MARK: - Private properties
-  
-  private var entries: [ObserverSetEntry<Parameters>] = []
-  
-  // MARK: - Initialisers
-  
-  /**
-   Creates a new instance of an observer set.
-   
-   - returns: A new instance of an observer set.
-   */
-  public init() {}
-  
-  // MARK: - Public functions
-  
-  /**
-   Adds a new observer to the set.
-   
-   - parameter observer: The object that is to be notified.
-   - parameter callBack: The function to call on the observer when the notification is to be delivered.
-   
-   - returns: An entry in the list of observers, which can be used later to remove the observer.
-   */
-  @discardableResult
-  open func add<ObserverType: AnyObject>(_ observer: ObserverType, _ callBack: @escaping (ObserverType) -> (Parameters) -> Void) -> ObserverSetEntry<Parameters> {
-    let entry = ObserverSetEntry<Parameters>(observer: observer) { observer in callBack(observer as! ObserverType) }
-    synchronized {
-      self.entries.append(entry)
-    }
-    return entry
-  }
-  
-  /**
-   Adds a new function to the list of functions to invoke when a notification is to be delivered.
-   
-   - parameter callBack: The function to call when the notification is to be delivered.
-   
-   - returns: An entry in the list of observers, which can be used later to remove the observer.
-   */
-  @discardableResult
-  open func add(_ callBack: @escaping (Parameters) -> Void) -> ObserverSetEntry<Parameters> {
-    return self.add(self) { ignored in callBack }
-  }
-  
-  /**
-   Removes an observer from the list, using the entry which was returned when adding.
-   
-   - parameter entry: An entry returned when adding a new observer.
-   */
-  open func remove(_ entry: ObserverSetEntry<Parameters>) {
-    synchronized {
-      self.entries = self.entries.filter{ $0 !== entry }
-    }
-  }
-  
+	fileprivate var isExpired: Bool { false }
```

Comment:
The current swift version for this repo is 4.0 so these one line implicit returns wont work yet. 

```suggestion
	fileprivate var isExpired: Bool { return false }
```